ts/docs/client.md

```diff
@@ -0,0 +1,86 @@
+# grpc.client
+
+`grpc.client` allows making any type of gRPC request and exposes a client for the request that can be used to send multiple messages and attach callbacks to the request's lifecycle.
+
+**The default transports are not capable of bi-directional streaming - this implementation exists to support future transports such as websockets that are capable of bi-directional streaming.**
+
+## API Docs:
+```typescript
+grpc.client(methodDescriptor: MethodDescriptor, props: ClientRpcOptions): Client;
+```
+
+`methodDescriptor` is a generated method definition ([see code generation for how to generate these](code-generation)).
+
+#### `ClientRpcOptions`:
+
+* `host: string`
+  * The server address (`"https://example.com:9100"`)
+* `transport?: TransportConstructor`
+  * (optional) A function to build a `Transport` that will be used for the request. If no transport is specified then a browser-compatible transport will be used. See [transport](transport).
+* `debug?: boolean`
+  * (optional) if `true`, debug information will be printed to the console
+
+#### `Client`:
+```typescript
+// Open the connection to the server
+start(metadata?: grpc.Metadata): void;
+
+// Send a single instance of the request message type
+send(message: grpc.ProtobufMessage): void;
+
+// Indicate to the server that the client has finished sending
+finishSend(): void;
```

Comment:
`finishSend()` is used by the client to indicate to the server that it's finished sending - it doesn't actually end the lifecycle of the rpc; in some usages the server only starts sending messages when the client indicates that it has finished sending.

I think naming this `end()` will make this less clear as it might be assumed that this function will just terminate the rpc.